lib/jsdom/living/nodes/HTMLInputElement-impl.js

```diff
@@ -590,6 +632,209 @@ class HTMLInputElementImpl extends HTMLElementImpl {
     return null;
   }
 
+  get _parsedValue() {
+    const converter = this._convertStringToNumber;
+    if (converter !== undefined) {
+      return converter(this.value);
+    }
+    return this.value;
+  }
+
+  // https://html.spec.whatwg.org/multipage/input.html#attr-input-step
+  get _step() {
+    let step = this._defaultStep;
+    if (this.hasAttribute("step") && this.getAttribute("step").toLowerCase() !== "any") {
+      const parsedStep = parseFloatingPointNumber(this.getAttribute("step"));
+      if (!isNaN(parsedStep) && parsedStep > 0) {
+        step = parsedStep;
+      }
+    }
+    return step;
+  }
+
+  // https://html.spec.whatwg.org/multipage/input.html#concept-input-step-scale
+  get _stepScaleFactor() {
+    const dayInMilliseconds = 24 * 60 * 60 * 1000;
+    switch (this.type) {
+      case "week":
+        return 7 * dayInMilliseconds;
+      case "date":
+        return dayInMilliseconds;
+      case "datetime-local":
+      case "datetime":
+      case "time":
+        return 1000;
+    }
+    return 1;
+  }
+
+  // https://html.spec.whatwg.org/multipage/input.html#concept-input-step-default
+  get _defaultStep() {
+    if (this.type === "datetime-local" || this.type === "datetime" || this.type === "time") {
+      return 60;
+    }
+    return 1;
+  }
+
+  // https://html.spec.whatwg.org/multipage/input.html#concept-input-min-zero
+  get _stepBase() {
+    const parseAttribute = attributeName => parseFloatingPointNumber(this.getAttribute(attributeName));
+    if (this.hasAttribute("min")) {
+      const min = parseAttribute("min");
+      if (!isNaN(min)) {
+        return min;
+      }
+    }
+    if (this.hasAttribute("value")) {
+      const value = parseAttribute("value");
+      if (!isNaN(value)) {
+        return value;
+      }
+    }
+    return this._defaultStepBase;
+  }
+
+  // https://html.spec.whatwg.org/multipage/input.html#concept-input-step-default-base
+  get _defaultStepBase() {
+    if (this.type === "week") {
+      // The start of week 1970-W01
+      return 259200000;
+    }
+    return 0;
+  }
+
+  _attributeApplies(attribute) {
+    return applicableTypesForAttribute[attribute].has(this.type);
+  }
+
+  get willValidate() {
+    // https://html.spec.whatwg.org/#hidden-state-(type=hidden)
```

Comment:
All URLs in comments should preferably go to the `/multipage` version of the spec. This applies to a few other files as well.